geopandas/tests/test_sindex.py

```diff
@@ -105,24 +122,52 @@ def setup_method(self):
     def test_sindex(self):
         self.df.crs = "epsg:4326"
         assert self.df.sindex.size == 5
-        with pytest.warns(FutureWarning, match="`objects` is deprecated"):
-            # TODO: remove warning check once deprecated
-            hits = list(self.df.sindex.intersection((2.5, 2.5, 4, 4), objects=True))
+        hits = list(self.df.sindex.intersection((2.5, 2.5, 4, 4)))
         assert len(hits) == 2
-        assert hits[0].object == 3
+        assert hits[0] == 3
 
     def test_lazy_build(self):
-        assert self.df._sindex is None
+        assert self.df.geometry.values._sindex is None
         assert self.df.sindex.size == 5
-        assert self.df._sindex is not None
+        assert self.df.geometry.values._sindex is not None
 
     def test_sindex_rebuild_on_set_geometry(self):
         # First build the sindex
         assert self.df.sindex is not None
+        original_index = self.df.sindex
         self.df.set_geometry(
             [Point(x, y) for x, y in zip(range(5, 10), range(5, 10))], inplace=True
         )
-        assert self.df._sindex_generated is False
+        assert self.df.sindex is not original_index
+
+    def test_rebuild_on_row_slice(self):
+        # Select a subset of rows rebuilds
+        original_index = self.df.sindex
+        sliced = self.df.iloc[:1]
+        assert sliced.sindex is not original_index
+        # Slicing all does not rebuild
+        original_index = self.df.sindex
+        sliced = self.df.iloc[:]
+        assert sliced.sindex is original_index
+        # Re-ordering rebuilds
+        sliced = self.df.iloc[::-1]
+        assert sliced.sindex is not original_index
+
+    @pytest.mark.skipif(
+        str(pd.__version__) <= LooseVersion("1.0.0"),
+        reason="Column selection returns a copy on pd<=1.0.0",
+    )
```

Comment:
@jorisvandenbossche could you confirm if this is the expected behavior, and if it's okay to just skip this test for old pandas versions (or if you want a test specifically for old pandas versions)? Thanks!

Reply:
I will check, but yeah, if it's failing for older pandas, this is fine to skip like this. 

You could only use `not compat.PANDAS_GE_10` for the version check

Reply:
Looking at the failures on Travis, this only seems to fail for the case of selecting multiple columns, right? In that case, I would maybe split the test in two: one test for selecting the geometry column (`df['geometry']` and `df.geometry`), and one test for selecting a subset of columns (`df[["col", "geometry"]]`), and then only the last one needs to be skipped.

And I can confirm with a small test that indeed with pandas < 1.0, selecting a subset of columns created a copy (not for selecting a single column)

Reply:
implemented. thanks!